src/Dialogs/Waypoint/WaypointCommandsWidget.cpp

```diff
@@ -202,78 +204,92 @@ ActivatePan(const Waypoint &waypoint)
   return PanTo(waypoint.location);
 }
 
+void
+WaypointCommandsWidget::UpdateButtons()
+{
+  has_freq = waypoint->radio_frequency.IsDefined();
+  SetRowEnabled(REPLACE_IN_TASK, task_manager != nullptr);
+  SetRowEnabled(INSERT_IN_TASK, task_manager != nullptr);
```

Comment:
INSERT_IN_TASK creates a new "goto task" currently. So this shouldn't be disabled if there is no current task defined.

Reply:
The same is true for APPEND_TO_TASK

Reply:
Ah I see. I think this is part of Max's defensive programming. Basically the task_manager is initialized to a pointer. If this pointer should be invalid for any reason, the button would be disabled.